src/cryptsetup/cryptsetup-generator.c

```diff
@@ -170,6 +170,19 @@ static int create_disk(
         swap = fstab_test_option(options, "swap\0");
         netdev = fstab_test_option(options, "_netdev\0");
 
+        c = strrchr(keyspec, ':');
+        if (c) {
+                password = strndup(keyspec, c-keyspec);
+                keydev = strdup(c + 1);
+                if (!password || !keydev)
+                        return log_oom();
+        } else {
+                /* No keydev specified */
+                password = strdup(keyspec);
+                if (!password)
+                        return log_oom();
+        }
```

Comment:
so I am not a fan of this, you split the keyspec string now at application time, not at parsing time. I think we should clearly do parsing when parsing and applying when applying but not mix both. Hence, please move this splitting to the parsing code, so that at this point we already have everything properly split up.